row.go

```diff
@@ -599,6 +625,7 @@ func reconstructRepeated(columnIndex, rowLength int16, levels levels, row Row, d
 
 	var err error
 	for row.startsWith(columnIndex) && row[0].repetitionLevel == levels.repetitionLevel {
+		// TODO: check specific error here.
```

Comment:
We should hopefully be getting a clear error that we're looking for to break the loop.  Feels dangerous to catch them all here otherwise.